include/nuraft_mesg/mesg_factory.hpp

```diff
@@ -38,16 +38,21 @@ using client_factory_lock_type = folly::SharedMutex;
 //   inherited rpc_client instances sharing a common worker pool.
 class grpc_factory : public nuraft::rpc_client_factory, public std::enable_shared_from_this< grpc_factory > {
     std::string _worker_name;
+    std::string _raft_worker_name;
+    std::string _data_worker_name;
 
 protected:
     client_factory_lock_type _client_lock;
     std::map< peer_id_t, std::shared_ptr< nuraft::rpc_client > > _clients;
 
 public:
     grpc_factory(int const cli_thread_count, std::string const& name);
+    grpc_factory(int const raft_cli_thread_count, int const data_cli_thread_count, std::string const& name);
     ~grpc_factory() override = default;
 
     std::string const& workerName() const { return _worker_name; }
+    std::string const& raftWorkerName() const { return _raft_worker_name; }
+    std::string const& dataWorkerName() const { return _data_worker_name; }
```

Comment:
Do we need to actually have two new member fields, can these accessories just return an inline `fmt` format around `workerName()` where it's called? It's just used at startup and shutdown yeah?

Reply:
I was not sure if we wanted to remove the support to have a single threadpool for both channels. I was thinking of using whether these member fields are set as a proxy to the threading model. (if `_worker_name` is set then its a single thread pool etc based on the constructor used). If this is not necessary, I can just removethe support for a single thread pool

Reply:
But it's not functional code (single pool) today right? I just don't want it to never be needed and this code lives on forever. Your choice.

Reply:
Yeah, no one is using it. I am for removing it completely, I do not like having a code that we will not use. I put it out in the initial draft to see how it looks like. If @hkadayam is fine, I will remove it.